jwt-authorizer/src/lib.rs

```diff
@@ -27,14 +26,16 @@ where
     T: DeserializeOwned + Send + Sync + Clone + 'static,
     S: Send + Sync,
 {
-    type Rejection = StatusCode;
+    type Rejection = AuthError;
 
     async fn from_request_parts(parts: &mut Parts, _: &S) -> Result<Self, Self::Rejection> {
-        if let Some(claims) = parts.extensions.get::<TokenData<T>>() {
-            Ok(JwtClaims(claims.claims.clone()))
-        } else {
-            tracing::error!("JwtClaims extractor must be behind a jwt-authoriser layer!");
-            Err(StatusCode::INTERNAL_SERVER_ERROR)
+        match parts.extensions.get::<Result<TokenData<T>, AuthError>>() {
+            Some(Ok(data)) => Ok(JwtClaims(data.claims.to_owned())),
+            Some(Err(e)) => Err(e.to_owned()),
+            None => {
```

Comment:
Good question! Axum supports `Option` in its extractors, however there is no distinction between extractors that failed because the supplied data is wrong, and those that failed because no data was supplied at all, so the `None` case represents 'missing and invalid' which is rarely what you want.

I have raised an issue in the past (https://github.com/tokio-rs/axum/issues/1883) and am working on a crate which I plan on finishing up / releasing this week that handles this case.

Perhaps it will be made possible by default in the future.

Reply:
Great work @arlyon - thank you 💜